### ISSUE TITLE
Significantly slower in latest version than in 0.4.0

### ISSUE THREAD

**Issue by YimengZhu** [pytorch/pytorch#24398]
Hi,

if I run the following script to train a model, one can aware significantly performance drawbacks in 1.2.0 than in 0.4.0.

```
import torch 
import torch.nn as nn
import torch.optim as optim
from torch.optim.lr_scheduler import MultiStepLR
import time 

class DummyModel(nn.Module):
    def __init__(self): 
        super(DummyModel, self).__init__() 
        
        self.conv1Ds= nn.Sequential(
            nn.Conv1d(60, 512, 5, padding=2), 
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Conv1d(512, 512, 3, padding=1),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Conv1d(512, 512, 3, dilation=2, padding=2),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Conv1d(512, 512, 3, dilation=4, padding=4),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Conv1d(512, 512, 3, dilation=6, padding=6),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Conv1d(512, 512, 1),
            nn.ReLU(),
            nn.BatchNorm1d(512)).cuda()

        self.linears = nn.Sequential(
            nn.Linear(512, 512),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Linear(512, 512),
            nn.ReLU(),
            nn.BatchNorm1d(512),

            nn.Linear(512, 30)).cuda()

    def forward(self, x):
        x = x.permute(0, 2, 1)
        x = self.conv1Ds(x)
        x = x.permute(0, 2, 1)
        x = self.linears(x)
        return x


def train():
    batch_size = 150
    feature = torch.rand(batch_size, 512, 60).cuda()
    label = torch.rand(batch_size, 30).long().cuda()

    criterion = nn.CrossEntropyLoss(ignore_index=-1).cuda()
    optimizer = optim.Adam(net.parameters(), lr=0.0005)

    for epoch in range(20):
        net.zero_grad()
        predict = net(feature)
        loss = criterion(predict, label)
        loss.backward()
        optimizer.step()


if __name__ == '__main__':
    print('torch version: {}'.format(torch.__version__))                                                                                                                                                                                         
    from_time = time.time()                                                                                                                                                                                                                      
    net = DummyModel()                                                                                                                                                                                                                           
    train()                                                                                                                                                                                                                                      
    running_time = time.time() - from_time                                                                                                                                                                                                       
    print('running time: {}'.format(running_time))
```
I'm not sure if this is a duplicate as [this one](https://github.com/pytorch/pytorch/issues/14456), a as according to [soumith](https://github.com/pytorch/pytorch/issues/15054#issuecomment-488482160), the newly installed pytorch should have the correct cudnn that fixed this issure. 

Config:
OS: Ubuntu 16.04.6 LTS
Cuda: 10.1
NVIDIA Driver Version: 418.56

**Comment by Lornatang** [comment]
I'm sorry, but I haven't found anything slow yet.

**Comment by soumith** [comment]
if it is significantly slow, please tell us what are the numbers you have on your end.

**Comment by YimengZhu** [comment]
> if it is significantly slow, please tell us what are the numbers you have on your end.

On 0.4.0, it is approximately 15s to run the script, whereas in 1.2.0 it is around 70s.

PS: this is an extremely simplified version of my training process, to which the problem is essentially related. In reality, the performance drawback can cause 1d vs 1week difference considering my dataset size.

**Comment by AlbertZhangHIT** [comment]
Hi, I also found that torch 1.2 is slower than torch 1.0.
I compared the runing performace of DCGAN-like (for 64x64 images) net on torch 1.2 and 1.0, for 100 iters torch 1.2 took about 90 seconds while torch 1.0 took 36 seconds. 
I use ```top```  command in linux finding that torch 1.2 uses CPU exceeding 1000% while torch 1.0 only uses 100%.
There should be a bug with the CPU and GPU utility in torch 1.2.

Thanks.

Config 1.
OS: Ubuntu 16.04.6 LTS
torch: 1.0.1
CUDA: 9.0
CUDNN: 7.0
GPU: TITAN V
NVIDIA Driver Version: 418.56

Config 2.
OS: Ubuntu 16.04.6 LTS
torch: 1.2
CUDA: 10.0
CUDNN: 7.5
GPU: TITAN V
NVIDIA Driver Version: 418.56

**Comment by soumith** [comment]
cc: @gchanan for scheduling

**Comment by shaswot-utokyo** [comment]
My code also seems much slower in pytorch 1.2 as compared to pytorch 0.4.
I ran the code on nvidia-docker containers.
Code implements the most basic DQN on Open AI Cartpole-v0 env.

Seems like CUDA version might have something to do with it. Will try to run the code on a pytorch 1.2, CUDA 9.2 container

Container 1:
torch version: 1.2.0.dev20190709
Cuda compilation tools, release 10.0, V10.0.130
RUNNING TIME: 102.240061044693
**********************************************************
Container 2:
torch version: 0.4.0
Cuda compilation tools, release 9.0, V9.0.176
RUNNING TIME: 62.04717779159546
**********************************************************

*****************************************************************************
CODE
*****************************************************************************
#!/usr/bin/env python

# Derived from https://github.com/higgsfield/RL-Adventure/blob/master/1.dqn.ipynb
# DQN without a frozen target network

import datetime
import time

seed_value = 324267

import math
import os 
import random 
import numpy as np 

import torch
import torch.nn as nn
import torch.optim as optim
import torch.autograd as autograd 
import torch.nn.functional as F


os.environ['PYTHONHASHSEED']=str(seed_value) 
random.seed(seed_value) 
np.random.seed(seed_value) 
torch.manual_seed(seed_value)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False


import gym
# CartPole-v0 Environment
env_id = "CartPole-v0"
env = gym.make(env_id)
env.seed(seed_value);

USE_GPU = False

# Use CUDA
USE_CUDA = torch.cuda.is_available() and USE_GPU

if USE_CUDA:
    torch.cuda.manual_seed(seed_value)
    device = torch.device('cuda')
else:
    device = torch.device('cpu')

# REPLAY BUFFER

from collections import deque

class ReplayBuffer(object):
    def __init__(self, capacity):
        self.buffer = deque(maxlen=capacity)
    
    def push(self, state, action, reward, next_state, done):
        state      = np.expand_dims(state, 0)
        next_state = np.expand_dims(next_state, 0)
            
        self.buffer.append((state, action, reward, next_state, done))
    
    def sample(self, batch_size):
        state, action, reward, next_state, done = zip(*random.sample(self.buffer, batch_size))
        return np.concatenate(state), action, reward, np.concatenate(next_state), done
    
    def __len__(self):
        return len(self.buffer)

class DQN(nn.Module): #base model
    def __init__(self, num_inputs, num_actions, HIDDEN_LAYER_WIDTH):
        super(DQN, self).__init__()
        
        self.action_dim = num_actions
        
        self.layers = nn.Sequential(
            nn.Linear(num_inputs, HIDDEN_LAYER_WIDTH),
            nn.ReLU(),
            nn.Linear(HIDDEN_LAYER_WIDTH, HIDDEN_LAYER_WIDTH),
            nn.ReLU(),
            nn.Linear(HIDDEN_LAYER_WIDTH, num_actions)
        )

    def forward(self, x):
        return self.layers(x)
    
    def act(self, state, epsilon):
        with torch.no_grad():
            if random.random() > epsilon:
                state   = torch.tensor(state, dtype=torch.float32).unsqueeze(dim=0).to(device)
                q_values = self.forward(state)
                action  = q_values.max(dim=1)[1].item()
            else:
                action = random.randrange(self.action_dim)
        return action

# e-greedy exploration
epsilon_start = 1.0
epsilon_final = 0.01
epsilon_decay = 500
epsilon_by_frame = lambda frame_idx: epsilon_final + (epsilon_start - epsilon_final) * math.exp(-1. * frame_idx / epsilon_decay)


# MODEL
# only one NN for estimating Q-values
model = DQN(env.observation_space.shape[0], 
            env.action_space.n,
            128)    
model = model.to(device)

optimizer = optim.Adam(model.parameters(),
                       lr=0.1)
criterion = nn.MSELoss()

# REPLAY BUFFER
replay_buffer = ReplayBuffer(capacity=1000)

def update_target(current_model, target_model):
    target.load_state_dict(model.state_dict())
    
def compute_td_loss(batch_size):
    state, action, reward, next_state, done = replay_buffer.sample(batch_size)

    state      = torch.tensor(np.float32(state)      ,dtype=torch.float32).to(device)
    next_state = torch.tensor(np.float32(next_state) ,dtype=torch.float32, requires_grad=False).to(device)
    action     = torch.tensor(action                ,dtype=torch.long).to(device)
    reward     = torch.tensor(reward                ,dtype=torch.float32).to(device)
    done       = torch.tensor(done                  ,dtype=torch.float32).to(device)

    q_values = model(state)
    q_value  = q_values.gather(dim=1, index=action.unsqueeze(dim=1)).squeeze(dim=1)

    next_q_values = model(next_state)
    next_q_value  = next_q_values.max(dim=1)[0]
    
    expected_q_value = reward + gamma * next_q_value * (1 - done)

    loss = criterion(q_value, expected_q_value.detach())
       
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()
    
    return loss.to('cpu')



if __name__ == '__main__':
    print('torch version: {}'.format(torch.__version__))                                                       
    from_time = time.time()                                                                                         
    # Training BEGINS
    num_frames = 50_000
    batch_size = 32
    gamma      = 0.9

    losses = []
    all_rewards = []
    episode_reward = 0

    state = env.reset()
    for frame_idx in range(1, num_frames + 1):
        epsilon = epsilon_by_frame(frame_idx)
        action = model.act(state, epsilon)

        next_state, reward, done, _ = env.step(action)
        replay_buffer.push(state, action, reward, next_state, done)

        state = next_state
        episode_reward += reward

        if done:
            state = env.reset()
            all_rewards.append(episode_reward)
            episode_reward = 0

        if len(replay_buffer) > batch_size:
            loss = compute_td_loss(batch_size)
    # Training ENDS
    
    running_time = time.time() - from_time
    print('running time: {}'.format(running_time))
*****************************************************************************

**Comment by DeqiangWang** [comment]
I also meet the same issue, I fine the pytoch doc(https://pytorch.org/docs/stable/notes/randomness.html) tell me to set :
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
but no use, and I try a lot time, find if I delete torch.backends.cudnn.deterministic = True and set torch.backends.cudnn.benchmark = True, get heighter training speed.....

**Comment by Jacobew** [comment]
@soumith Hi, any updates here? I've got a similar issue when using depthwise separable convolution. When torch.backends.cudnn.deterministic is set to be True, the training time of an iteration reduces from 2 min to 1 min. And actually setting torch.backends.cudnn.benchmark = True will slow down training a little bit... I found it quite different from PyTorch documents. 
My CUDA version:10.1, Pytorch version:1.0.0

**Comment by XiaolongMeng** [comment]
same issue when using pytorch
when version is 1.3 and use cudnn ,usage of gpu is only 17%,
when version is 1.3,it's 30%,
when version is 0.4, it's 60%,
so what's up?
CUDA version:10.1 , GPU is P100

**Comment by Dewald928** [comment]
I've found that changing to pytorch 1.2 (haven't tried 1.3) with cudatoolkit 10.0. 
And setting `torch.backends.cudnn.deterministic = False` I've had the best training times,
but naturally at a cost of reproducibility and determinism.

**Comment by rose-jinyang** [comment]
Hello
I met the same kind of issue.
I trained two models with the same architecture and dataset on PyTorch 0.4.1 and any later version.
The model trained on PyTorch 0.4.1 can forward in 200ms.
But the model trained on a later version of PyTorch can forward in 400ms.
I repeated many experiments but the results are the same.
What is the reason?
